Lib/pprint.py

```diff
@@ -176,8 +177,22 @@ def _format(self, object, stream, indent, allowance, context, level):
                 p(self, object, stream, indent, allowance, context, level + 1)
                 del context[objid]
                 return
+            elif _dataclasses.is_dataclass(object) and not isinstance(object, type):
+                context[objid] = 1
+                self._pprint_dataclass(object, stream, indent, allowance, context, level + 1)
+                del context[objid]
+                return
         stream.write(rep)
 
+    def _pprint_dataclass(self, object, stream, indent, allowance, context, level):
+        cls_name = object.__class__.__name__
+        indent += len(cls_name) + 1
+        items = [(f.name, getattr(object, f.name)) for f in _dataclasses.fields(object)]
```

Comment:
I think you should ignore fields where `f.repr` is false.

Reply:
Agreed. Similarly, I've opted not to do special formatting for dataclasses created with `repr=False` (otherwise we'd need to move the check for whether the object is a dataclass above the check for whether the default repr is short enough, which feels like the wrong change to be making).